src/Orleans.Storage.Redis/Providers/Storage/RedisGrainStorage.cs

```diff
@@ -35,26 +35,27 @@ public class RedisGrainStorage : IGrainStorage, ILifecycleParticipant<ISiloLifec
             NullValueHandling = NullValueHandling.Ignore
         };
 
-        private ConnectionMultiplexer _redisConnection;
-        private IDatabase _redisClient => _redisConnection.GetDatabase();
+        private IConnectionMultiplexerFactory _connectionMultiplexerFactory;
+        private IConnectionMultiplexer _connectionMultiplexer;
+        private IDatabase _redisClient => _connectionMultiplexer.GetDatabase();
 
-        public RedisGrainStorage(string name, RedisGrainStorageOptions options, ILogger logger, IOptions<ClusterOptions> clusterOptions, ISerializationManager serializationManager)
+        public RedisGrainStorage(string name, RedisGrainStorageOptions options, ILogger logger, IOptions<ClusterOptions> clusterOptions, ISerializationManager serializationManager, IConnectionMultiplexerFactory connectionMultiplexerFactory)
         {
             _name = name;
             _options = options;
             _logger = logger ?? SilentLogger.Logger;
             _serializationManager = serializationManager;
             _clusterOptions = clusterOptions.Value;
+            _connectionMultiplexerFactory = connectionMultiplexerFactory;
         }
 
         public async Task Init(CancellationToken ct)
         {
-            _redisConnection = await ConnectionMultiplexer.ConnectAsync(_options.ConnectionString);
+            _connectionMultiplexer = await _connectionMultiplexerFactory.CreateAsync(_options.ConnectionString);
         }
 
         public Task Close(CancellationToken ct)
         {
-            _redisConnection.Dispose();
```

Comment:
Does the connection ever get disposed anywhere?

Reply:
Nope - realistically it shouldn't until the host is brought down.

Reply:
Since we always do b/g in our real stacks then this wouldn't be a problem there, but if the Redis connection uses unmanaged system resources then I could see this being a problem for local development where killing the process may not free up everything on the system.  It's probably not urgent, but it is worth looking into at some point.

Reply:
#6